### PR TITLE
Feature/volunteers activation

### PR DESCRIPTION
**Pull request by pavlokolodka**
https://trello.com/c/ND4SotxV

### CODE REVIEW

**Comment by pavlokolodka** [comment]
⚠️ Merge with https://github.com/CHI-internship/KraudDonate-Service/pull/45

File: src/api/admin.api.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      localStorage.setItem(
+        'token',
+        admin.token
+      );
+    }  
+    return admin;  
+  }
+
+  async getVolunteersRequests() {
+    const requests = await axiosInstance
```
Comment (RamboUA) [inline src/api/admin.api.js:19]: return it directly without await
```suggestion
    return axiosInstance
```

File: src/api/admin.api.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      .get('admin/requests')
+      .then(data => data.data) 
+
+    return requests;  
+  }
+
+  async getVolunteerRequest(id) {
+    const request = await axiosInstance
```
Comment (RamboUA) [inline src/api/admin.api.js:27]: same

File: src/api/admin.api.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      .get(`admin/requests/${id}`)
+      .then(data => data.data) 
+
+    return request;  
+  }
+
+  async approveRequest(userId, status, message = null) {
```
Comment (RamboUA) [inline src/api/admin.api.js:34]: same

File: src/components/Request.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  const [request, SetRequest] = useState({});
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    getRequest(SetRequest);
+  }, [])
+  
+  const extension = request?.document?.split('.').pop();
+
+  return (
```
Comment (RamboUA) [inline src/components/Request.js:29]: Something wrong with indents, please check hierarchy

Comment (RamboUA) [reply src/components/Request.js:29]: And remove blank lines, please

File: src/components/Request.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      </div>
+  
+      {extension === 'pdf' &&
+        <iframe height="400" width="100%" src={"https://docs.google.com/gview?url=" + request.document + "&embedded=true"}></iframe>
+      }
+      <TextField id="reject" label="Input reject reason" variant="standard" sx={{ marginLeft: "20%"}}/>
+    </CardContent>
+    <CardActions>
+      <Button size="small" variant="contained"
+      onClick={async () => {
```
Comment (RamboUA) [inline src/components/Request.js:68]: Let's move this handler to a function outside

File: src/components/Request.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        await adminService.approveRequest(request.userId, true);
+        navigate('/admin/requests'); 
+      }}
+      >
+        Approve
+      </Button>
+      <Button size="small" variant="contained"
+      onClick={async () => {
```
Comment (RamboUA) [inline src/components/Request.js:76]: Same

File: src/components/Requests.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import { useNavigate } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import adminService from '../api/admin.api'
+
+const getRequests = async (setState) => {
+  const requests = await adminService.getVolunteersRequests();
+  setState(requests);
+}
+
+const getRequest = async (id) => {
```
Comment (RamboUA) [inline src/components/Requests.js:16]:
```suggestion
const getRequest = async (id) =>  adminService.getVolunteerRequest(id);
```

File: src/components/Requests.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+export default function Requests() {
+  const navigate = useNavigate();
+  const [requests, SetRequests] = useState([]);
+
+  useEffect(() => {
+    getRequests(SetRequests);
+  }, [])
+  
+  return (
```
Comment (RamboUA) [inline src/components/Requests.js:29]: Please fix indents hierarchically  and move onClick handlers to a function outside

**Review by RamboUA — CHANGES_REQUESTED** [review]
Thank you, please check the comments

File: src/components/Requests.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+            <Button size="small" onClick={async () => await handleApprove(element.id)}>
+              Approve
+            </Button>
+            <Button size="small"
+            onClick={() => navigate(`/admin/request/${element.id}`)}>
+              Learn More
+            </Button>
+          </CardActions>
+      </Card>
+    )
```
Comment (RamboUA) [inline src/components/Requests.js:54]:
```suggestion
    )}
```
Move it to single line, please

File: src/components/SignIn.js [hunk 10]
```diff
@@ -8,25 +8,25 @@ import {
...
             email: '',
             password: '',
         },
         validationSchema: validationSchema,
         onSubmit: async values => {
-            console.log({
-                email: values.email,
-                password: values.password,
-            });
+            await adminService.login(values)
```
Comment (RamboUA) [inline src/components/SignIn.js:28]: I think async/await is not necessary here. Check it please

File: src/components/Requests.js [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+            <Typography gutterBottom variant="h5" component="div">
+              Activate Request
+            </Typography>
+            <Typography variant="body1" color="text.secondary">
+              Country - {element.country} <br/>
+              Card number - {element.card_number}
+            </Typography>
+          </CardContent>
+          <CardActions>
+            <Button size="small" onClick={async () => await handleApprove(element.id)}>
```
Comment (RamboUA) [inline src/components/Requests.js:45]: should work without async/await

**Review by RamboUA — CHANGES_REQUESTED** [review]

**Pull request merged**